add help menu to install.sh

qubenix opened the pull request:
I tried to run the install script yesterday without first checking the readme, and I ended up having to look into the code to find out what options I could add. This PR adds a help menu to the install script which will be presented to the user any time they add any flag which is not supported (including `-h` or `--help`).

I removed the error message about "unknown flag" and replaced it with this menu. IMO this is more helpful. ~~I added an `exit`  after the help menu which some may prefer as `exit 1`.~~

Whoops, looks like I left it `exit 1`. Happy to change it to whichever is preferred.

kristapsk commented on install.sh:
```diff
@@ -356,11 +356,9 @@ parse_flags ()
-                ;;
             *)
-                break
+                echo -e "\nUsage: "${0}" [options]\n\nOptions:\n\n--develop       code remains editable in place\n--python, -p    python version: python2 or python3 (default: python2)\n--with-qt       build the Qt GUI (incompatible with python2)\n"
```
--python/-p allows not only python2/python3, but also more specific versions like python2.7 or python3.6 (was necessary for my environment).

qubenix replied:
Thanks for pointing that out. I've removed the options for `-p` and just left the default as an example. If someone has a better suggestion just let me know.

I'll squash the commits once I'm sure that there's no more to be made.

fivepiece commented on install.sh:
```diff
@@ -356,11 +356,9 @@ parse_flags ()
-                ;;
             *)
-                break
+                echo -e "\nUsage: "${0}" [options]\n\nOptions:\n\n--develop       code remains editable in place\n--python, -p    python version (default: python2)\n--with-qt       build the Qt GUI (incompatible with python2)\n"
```
No strong feelings about this, but if you preformat the thing printed by `echo` like
```
echo "
    Usage ...
    option1, info option1
    option2, info option2
    ...
```
Then adding or removing something will be an easier diff to look at instead of the whole line changing

kristapsk replied:
I agree with this.

qubenix commented:
@fivepiece Thanks for reviewing and your suggestions. I think I've addressed all those concerns.

I also like to preformatted the menu, I just wasn't sure if breaking indentation was frowned on or not.

AdamISZ commented:
LGTM as general idea, I'll take @fivepiece 's approval on the technical aspect.
